### PR TITLE
trezor-core-tools python package

### PR DESCRIPTION
**Pull request by matejcik**
this, essentially, replaces `./core/tools/headertool.py` with just `headertool`

also improves `combine_firmware` so that it (a) accepts model name as an argument, instead of path to the header file, (b) accepts lowercase section names, (c) is friendlier when you mess up a section name

### CODE REVIEW

File: core/tools/trezor_core_tools/common.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+HERE = Path(__file__).parent.resolve()
+CORE = HERE.parent.parent
+
+MODELS_DIR = CORE / "embed" / "models"
+
+MODELS_DICT = {
```
[the remark this answers is not in the document]
Comment (matejcik) [reply core/tools/trezor_core_tools/common.py:10]: 965cded3187b5186d509077dd261693badc3cfdd
ditching the names is a task for another day ... with that i'd also ditch `T` and `R`

File: core/tools/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+TBD
```
Comment (obrusvit) [inline core/tools/README.md:1]: unfinished

File: core/Makefile [hunk 3]
```diff
@@ -46,66 +46,59 @@ OPENOCD_TRANSPORT ?= hla_swd
...
-BOOTLOADER_SECTOR_END    = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} BOOTLOADER_SECTOR_END)
-FIRMWARE_SECTOR_START    = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} FIRMWARE_SECTOR_START)
-FIRMWARE_SECTOR_END      = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} FIRMWARE_SECTOR_END)
-FIRMWARE_P2_SECTOR_START = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} FIRMWARE_P2_SECTOR_START)
-FIRMWARE_P2_SECTOR_END   = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} FIRMWARE_P2_SECTOR_END)
-STORAGE_1_SECTOR_START   = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} STORAGE_1_SECTOR_START)
-STORAGE_1_SECTOR_END     = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} STORAGE_1_SECTOR_END)
-STORAGE_2_SECTOR_START   = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} STORAGE_2_SECTOR_START)
-STORAGE_2_SECTOR_END     = $(shell python ./tools/layout_parser.py ${LAYOUT_FILE} STORAGE_2_SECTOR_END)
+FLASH_START              = $(shell layout_parser ${TREZOR_MODEL} FLASH_START)
```
Comment (obrusvit) [inline core/Makefile:77]: `FLASH_START` seems to be unused.

File: core/Makefile [hunk 4]
```diff
@@ -46,66 +46,59 @@ OPENOCD_TRANSPORT ?= hla_swd
...
+STORAGE_1_OFFSET         = $(shell layout_parser ${TREZOR_MODEL} STORAGE_1_OFFSET)
+STORAGE_2_OFFSET         = $(shell layout_parser ${TREZOR_MODEL} STORAGE_2_OFFSET)
+STORAGE_SIZE             = $(shell layout_parser ${TREZOR_MODEL} NORCOW_SECTOR_SIZE)
+BOARDLOADER_MAXSIZE      = $(shell layout_parser ${TREZOR_MODEL} BOARDLOADER_IMAGE_MAXSIZE)
+BOOTLOADER_MAXSIZE       = $(shell layout_parser ${TREZOR_MODEL} BOOTLOADER_IMAGE_MAXSIZE)
+FIRMWARE_MAXSIZE         = $(shell layout_parser ${TREZOR_MODEL} FIRMWARE_IMAGE_MAXSIZE)
+FIRMWARE_P1_MAXSIZE      = $(shell layout_parser ${TREZOR_MODEL} FIRMWARE_P1_IMAGE_MAXSIZE)
+FIRMWARE_P2_MAXSIZE      = $(shell layout_parser ${TREZOR_MODEL} FIRMWARE_P2_IMAGE_MAXSIZE)
+BOARDLOADER_SECTOR_START = $(shell layout_parser ${TREZOR_MODEL} BOARDLOADER_SECTOR_START)
+BOARDLOADER_SECTOR_END   = $(shell layout_parser ${TREZOR_MODEL} BOARDLOADER_SECTOR_END)
```
Comment (obrusvit) [inline core/Makefile:91]: `BOARDLOADER_SECTOR_START` and `BOARDLOADER_SECTOR_END` seem to be unused.

**Review by obrusvit — COMMENTED** [review]
I was able to use new tools `layout_parser` and `headertool` after invoking `poetry install` from this branch. 

Apart from these minor issues, seems to be OK.

File: core/tools/trezor_core_tools/layout_parser.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import re
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (matejcik) [reply core/tools/trezor_core_tools/layout_parser.py]: e68188d232ea643ee90c6ed28a5f90f882d3e05e

File: core/tools/trezor_core_tools/layout_parser.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+
+# the following regular expression matches a thing looking like those examples:
+#   #define HEADER_START 0x123456
+#   #define HEADER_START (1 * 2 * 3) // comment
+# and returns two groups: the name and the value. Comment is ignored.
+SEARCH_PATTERN = r"#define (\w+) (.+?)(?:\s*//.*)?$"
+
+
+def extract_value(line: str, def_name: str) -> int | None:
```
[the remark this answers is not in the document]
Comment (matejcik) [reply core/tools/trezor_core_tools/layout_parser.py]: e68188d232ea643ee90c6ed28a5f90f882d3e05e

File: core/tools/README.md (hunk 2 above)
Comment (matejcik) [reply core/tools/README.md]: e2e4e999dbec0efa1f2c4eebe8be9c9d4735b6f3

File: core/Makefile (hunk 3 above)
Comment (matejcik) [reply core/Makefile:77]: i'd keep it around in case something like `make flash` would use it? wdyt @TychoVrahe

File: core/Makefile (hunk 4 above)
Comment (matejcik) [reply core/Makefile:91]: dtto

**Review by obrusvit — APPROVED** [review]

**Pull request merged**